lib/aws-manager.js

```diff
@@ -1247,7 +1125,9 @@ class AwsManager {
 
     let perRegionKills = {};
 
-    for (let region of this.ec2.regions) {
+    let killPromises = [];
+
+    for (let region of _.keys(this.ec2)) {
```

Comment:
I like the idea of SmartFilter, but I think we can get just as clear results with what I've changed to doing:

```javascript
      let instances = this.__apiState.instances.filter(instance => {
        return instance.WorkerType === name && states.includes(instance.State.Name); 
      });
```